integreat_cms/static/src/js/pages/xliff-export-overlay.ts

```diff
@@ -0,0 +1,83 @@
+  document.getElementById("xliff-overlay-bulk-action-execute")?.addEventListener("click", () => {
+    document.getElementById("bulk-action-execute").click();
+  });
+  // Set listener for select toggles.
+  document.querySelectorAll(".bulk-select-language").forEach((el) => {
+    el.addEventListener("change", toggleBulkActionButton); 
+  }); 
+  // Set listener for select-all toggle.
+  document.getElementById("select-all-languages")?.addEventListener("change", (toggle) => {
+    document.querySelectorAll(".bulk-select-language").forEach((el) => {
+      (el as HTMLInputElement).checked = (toggle.target as HTMLInputElement).checked; 
+      toggleBulkActionButton(); 
+    });
+  });
+});
+
+
+function addXliffExportOverlayListeners(callback: (overlay: HTMLElement) => void) {
+  const overlay = document.getElementById("xliff_export_overlay");
+  if (overlay) {
+    document.querySelectorAll("[data-show-xliff-export-overlay]").forEach((btn) => {
+      if (!btn.hasAttribute("data-has-listener")) {
+        btn.addEventListener("click", () => {
+          callback(overlay)
+        });
+        btn.setAttribute("data-has-listener", "");
+      }
+    });
+    document.getElementById("btn-close-xliff-export").addEventListener("click", () => {
+      closeXliffExportOverlay(overlay);
+    });
+    // Close window by clicking on backdrop.
+    overlay.addEventListener("click", (e) => {
+      if (e.target === overlay) {
+        closeXliffExportOverlay(overlay);
+      }
+    });
+  }
+}
+
+function openXliffExportOverlay(overlay: HTMLElement) {
+  overlay.classList.remove("hidden");
+  overlay.classList.add("flex");
+}
+
+
+function closeXliffExportOverlay(overlay: HTMLElement) {
+  overlay.classList.add("hidden");
+  overlay.classList.remove("flex");
+  let selector = (document.getElementById("bulk-action") as HTMLSelectElement);
+  if (selector.hasAttribute("data-last-value")) {
+    selector.value = selector.getAttribute("data-last-value");
+    selector.setAttribute("data-value", selector.value); 
+  }
+  toggleBulkActionButton();
+}
+
+
+function addSelectionTrackerListener() {
+  /* To give the bulk action for multi language XLIFF export a more button-like
+   feeling (as it opens a pop-up menu), this function implements an selection
+   tracker, so that the selection will never save this option, but jump back
+   to the last option value, when action is canceled. */
+  let selector = document.getElementById("bulk-action") as HTMLSelectElement;
+  if (selector) {
+    selector.setAttribute("data-value", selector.value);
+    selector.setAttribute("data-last-value", "");
+    selector.addEventListener("change", () => {
+      let val = selector.getAttribute("data-value");
+      let lastVal = selector.getAttribute("data-last-value");
+        selector.setAttribute("data-value", selector.value); 
+      if (val != lastVal) {
+        selector.setAttribute("data-last-value", val); 
+      }
+    });
+  }
+}
```

Comment:
I don't understand the JS part completely to be honest and feel like it is more complicated than needed, but other than that: Could you open the selection window only when the action is actually executed, not when the option is selected from the dropdown?
(I guess this would also solve the problem that it's currently possible to open the overlay when no pages are selected at all and thus the multi lang export itself cannot be executed....)